### DISCUSSION TITLE
Speed up reading mesh file?

### DISCUSSION THREAD

**Discussion by reed-lfh** [idaholab/moose#24360] [Q&A Meshing] [answered]
Hi everyone,

Recently, I have been trying to let the MOOSE read a pretty big .msh file (about 1.2 Gb). The MOOSE needs plenty of hours (>30hrs) to finish constructing the mesh as shown below. For this mesh file, there are different blocks that show different Euler angles. I am wondering why this is happening and whether there is a more efficient way of handling the mesh like this.

Thanks.

Mesh..............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

**Comment by GiudGiud** [comment, marked as answer]
Hello

Are you reading this in serial?
Did you ever get it to completion?
Do you have the option to convert it to another supported mesh format?
https://mooseframework.inl.gov/source/meshgenerators/FileMeshGenerator.html

Guillaume

**Reply by reed-lfh** [reply]
Hi Guillaume,

I am using mpi in sending this job. But I am not sure whether the MOOSE is reading it in parallel. Because you are right, I find I can hardly make it completed. 

I think it can be saved in another format. But I am not sure which one can be optimal.

Reed

**Reply by GiudGiud** [reply]
Have you had it read in serial once?

**Reply by GiudGiud** [reply]
if you have, you can save it to exodus from MOOSE. Reading exodus should pose no issue at all and be quite fast

**Reply by reed-lfh** [reply]
I believe I knew what happened. I just tried to run it in serial and I found it looks like I requested just so limited memory in reading this mesh file. But for some reason, the console still shows it was dealing with the mesh. Maybe there is something that can be improved here.

Thanks for the help!
